it/topics/scripting_languages/JavaScript/node_packages.md

```diff
@@ -0,0 +1,218 @@
+
+Passaggi di esempio dal flusso di lavoro di esempio di azioni Github:
+
+```Yaml
+    passaggi:
+    # ...
+    - esegui: npm ci
+    - run: npm run build --if-present
+    - esegui: test npm
+```
+
+
+Nota: richiede il file `package-lock.json` o il pacchetto termoretraibile. Non funziona con `yarn.lock`.
+
+Azioni Github: [Setup Node.js Environment](https://github.com/marketplace/actions/setup-node-js-environment)
+
+
+## Yarn
+
+### Caratteristiche
+
+- Può installare pacchetti dalla cache locale.
+- Vincola fortemente le versioni del pacchetto.
+- Installazione di pacchetti paralleli.
+- Comunità di utenti attivi.
+
+### Come usarlo
+
+Yarn è costruito su NPM, quindi per lo più puoi trattarlo allo stesso modo.
+
+- Richiede il comando `Yarn` per essere installato a livello globale o nel progetto, usando NPM.
+- Utilizza lo stesso `package.json` di NPM.
+- Usa `yarn.lock` invece di` package.lock.json`.
+- I comandi NPM familiari come `install` e` start` funzionano ancora.
+
+### Risorse
+
+- [yarnpkg.com](https://yarnpkg.com)
+    > Progetti sicuri, stabili e riproducibili
+    >
+    > Yarn è un gestore di pacchetti che funge anche da project manager. Sia che tu lavori su progetti one-shot o grandi monorepos, come hobbista o utente aziendale, ti abbiamo coperto.
+- [Installa Yarn](https://classic.yarnpkg.com/en/docs/install)
+- Tutorial
+    - [Perché e come migrare da NPM a YARN](https://waverleysoftware.com/blog/yarn-vs-npm/)
+
+
+### Yarn CI
+
+[Github Action for Yarn](https://github.com/marketplace/actions/github-action-for-yarn)
+
+
+## NVM
+
+Se si desidera installare più versioni di Node e NPM sul proprio computer, considerare l'utilizzo di NVM. 
+
+Questo non sostituisce `npm` ma consente di alternare tra le versioni di` npm`. Alcuni progetti potrebbero non funzionare con una versione precedente o più recente di `node` o` npm`.
+
+È possibile passare facilmente da una versione all'altra di Node e installare i pacchetti in un ambiente NPM specifico.
+
+Se si esegue `npm install PACKAGE` con una determinata versione, i pacchetti andranno qui per esempio (sui sistemi Unix).
+
+```
+~/.nvm/version/node/v8.9.0/lib/node_modules
+```
+
+### Impostare
+
+Segui le istruzioni qui:
+
+- [nvm-sh / nvm](https://github.com/nvm-sh/nvm) repository Github
+- [NVM, il modo più semplice per passare da un ambiente all'altro Node.js sulla tua macchina](https://itnext.io/nvm-the-easiest-way-to-switch-node-js-environments-on-your -machine-in-a-flash-17babb7d5f1b? gi = 74712a4b1ad)
+
+Nota che dovrai installare il comando `nvm` e anche aggiornare il tuo file RC della shell per assicurarti che venga prelevato.
+
+### Utilizzo
+
+Dopo l'installazione, è possibile utilizzarlo per installare e passare a una versione di Node di destinazione (e alla versione NPM associata).
+
+Ad esempio, qui installo una nuova versione di Node, senza specificare la versione completa.
+
+```Sh
+$ nvm install 13
+Download e installazione di Node v13.11.0...
+...
```

Comment:
Output that I pasted from the terminal comes directly from the tool so does not need to be translated.